5.0/fr/0x20-V11-Cryptography.md

```diff
@@ -0,0 +1,107 @@
+
+## V11.5 Valeurs aléatoires
+
+La génération de nombres pseudo-aléatoires cryptographiquement sécurisée (CSPRNG) est extrêmement difficile à mettre en œuvre. En général, les bonnes sources d'entropie d'un système s'épuisent rapidement si elles sont surexploitées, tandis que des sources moins aléatoires peuvent conduire à des clés et des secrets prévisibles.
+
+| # | Description | Niveau |
+| :---: | :--- | :---: |
+| **11.5.1** | Vérifiez que tous les nombres et chaînes aléatoires destinés à être non devinables doivent être générés à l'aide d'un générateur de nombres pseudo-aléatoires cryptographiquement sécurisé (CSPRNG) et posséder au moins 128 bits d'entropie. Notez que les UUID ne respectent pas cette condition. | 2 |
+| **11.5.2** | Vérifiez que le mécanisme de génération de nombres aléatoires utilisé est conçu pour fonctionner en toute sécurité, même en cas de forte demande. | 3 |
+
+## V11.6 Cryptographie à clé publique
+
+La cryptographie à clé publique sera utilisée lorsqu'il est impossible ou non souhaitable de partager une clé secrète entre plusieurs parties.
+
+Dans ce contexte, des mécanismes d'échange de clés approuvés, tels que Diffie-Hellman et Elliptic Curve Diffie-Hellman (ECDH), sont nécessaires pour garantir la sécurité du cryptosystème face aux menaces modernes. Le chapitre « Communication sécurisée » décrit les exigences pour TLS. Les exigences de cette section sont donc destinées aux situations où la cryptographie à clé publique est utilisée dans des cas d'utilisation autres que TLS.
+
+| # | Description | Niveau |
+| :---: | :--- | :---: |
+| **11.6.1** | Vérifier que seuls des algorithmes cryptographiques et des modes opératoires approuvés sont utilisés pour la génération et l'amorçage des clés, ainsi que pour la génération et la vérification des signatures numériques. Les algorithmes de génération de clés ne doivent pas générer de clés non sécurisées et vulnérables aux attaques connues, par exemple les clés RSA vulnérables à la factorisation de Fermat. | 2 |
+| **11.6.2** | Vérifiez que des algorithmes cryptographiques approuvés sont utilisés pour l'échange de clés (tels que Diffie-Hellman), en veillant à ce que les mécanismes d'échange de clés utilisent des paramètres sécurisés. Cela permettra d'éviter les attaques contre le processus d'établissement des clés, susceptibles de conduire à des attaques de type « adversaire du milieu » ou à des failles cryptographiques. | 3 |
+
+## V11.7 Cryptographie des données en cours d'utilisation
+
+La protection des données pendant leur traitement est primordiale. Des techniques telles que le chiffrement intégral de la mémoire, le chiffrement des données en transit et la garantie d'un chiffrement des données le plus rapide possible après utilisation sont recommandées.
```

Comment:
"et la garantie d'un chiffrement des données le plus rapide possible après utilisation sont recommandées." → I think we can understand this backwards. I would say "et s'assurer que le les données sont chiffrés le plus tôt possible après utilisation"